src/main/java/prefs/ConfigFileHandler.java

```diff
@@ -12,67 +12,131 @@
+ * cfhGlobal.saveConfig();
+ */
 public class ConfigFileHandler {
 
     private static final Logger logger = LogManager.getLogger(ConfigFileHandler.class.getName());
 
     private static final String CHARSET = "UTF-8";
-    private final String globalConfigFileName;
+    private final String configFileName;
     private final String configDirectory;
 
+    private Config config;
+
     private Gson gson;
 
-    public ConfigFileHandler(String configDirectory, String globalConfigFileName) {
-        this.globalConfigFileName = globalConfigFileName;
+    /**
+     * Constructs a new Config File Handler that handles a specific Config type.
+     * @param configDirectory The directory that the config file is held in
+     * @param configFileName The file name of the config file
+     * @param config The config file object to deserialise into.
+     */
+    public ConfigFileHandler(String configDirectory, String configFileName, Config config) {
+        this.configFileName = configFileName;
         this.configDirectory = configDirectory;
+        this.config = config;
 
         setupGson();
         ensureDirectoryExists();
     }
 
-    private void ensureDirectoryExists() {
+    /**
+     * Ensures that the directory specified in the attribute exists.
+     * Attempts to create the directory if it does not exist.
+     * @return True if the directory exists or was created.
+     *         False if by the end of the method, the directory still doesn't exist
+     */
+    private boolean ensureDirectoryExists() {
         File directory = new File(configDirectory);
-        boolean directoryNonExistent = !directory.exists() || !directory.isDirectory();
-        boolean exists = true;
+        boolean exists = directory.exists();
+        boolean isDir = directory.isDirectory();
+        boolean directoryNonExistent = !exists || !isDir;
         if (directoryNonExistent) {
             exists = directory.mkdirs();
         }
         if (!exists) {
             logger.warn("Could not create config file directory");
         }
+        return exists;
     }
 
-    private File getGlobalConfigFile() {
-        return new File(configDirectory, globalConfigFileName).getAbsoluteFile();
+    /**
+     * Returns the File representation at the config file path
+     */
+    private File getConfigFile() {
+        return new File(configDirectory, configFileName).getAbsoluteFile();
     }
 
-    public void saveGlobalConfig(GlobalConfig config) {
-        File configFile = getGlobalConfigFile();
+    /**
+     * Serialises the config attribute object to a config file.
+     */
+    public void saveConfig() {
+        File configFile = getConfigFile();
         try {
             Writer writer = new OutputStreamWriter(new FileOutputStream(configFile) , CHARSET);
-            gson.toJson(config, GlobalConfig.class, writer);
+            if (config == null) {
```

Comment:
Is there any case where the config can be null? If not, we should handle it as an error instead in the constructor or something. If there is any case where it can be null, then we should use `Optional`.